lib/datasource.js

```diff
@@ -2594,6 +2594,74 @@ DataSource.prototype.ping = function(cb) {
   return cb.promise;
 };
 
+/**
+ * Execute an arbitrary command. The commands are connector specific,
+ * please refer to the documentation of your connector for more details.
+ *
+ * @param command String|Object The command to execute, e.g. an SQL query.
+ * @param [args] Array Parameters values to set in the command.
+ * @param [options] Object Additional options, e.g. the transaction to use.
+ * @param [callback] Function The callback to call when done.
+ * @promise
+ */
+DataSource.prototype.execute = function(command, args, options, cb) {
```

Comment:
Considering that this new API will be available for LB4 apps only (no 2.x/3.x), do we still need to keep supporting callback style?

I would prefer to provide Promise mode only, it will greatly simplify the implementation.

Of course, connector's `execute` API is staying callback-based, that's not going to change.